### ISSUE TITLE
FPPD occasionally dies with Remote Falcon and MQTT Publish enabled

### ISSUE THREAD

**Issue by weazel91** [FalconChristmas/fpp#916]
**FPP Version and Hardware**: 
FPP V 4.5.1
Remote Falcon V 5.1.5

**Describe the bug and Steps to reproduce**: 
When using remote falcon in voting mode and requesting a vote FPP will play the requested vote perfectly. However on return to the main scheduled playlist, OCCASIONALLY FPPD dies and will restart. This means that viewers see the first song in the main playlist very often, so much so that Remote Falcon is required to be disabled at times as 'Into the Unknown' (the first song in my show) will play after every one or two votes at times.

**Expected behavior**: 
After FPP is done with the remote falcon sequence in the remote falcon playlist FPP should then return to the main playlist, at the point where it was prior to the request coming in. I.e. Main Playlist index 5 -> Remote Falcon Sequence -> Main Playlist index 6. This does occasionally work fine, but other times randomly does not.

**Additional context**: 
The video attached in the ZIP shows exactly the issue. Timings to describe the issue are below:

00:00 - FPP Playing main sequence
00:00 (SINGLE FRAME) - Request of 'Christmas Eve - Trans Siberian Orchestra' is picked up my Remote Falcon
00:07 - FPP Plays the remote falcon request as expected
00:25 - Request song ends
00:26 - FPPD DIES (unexpected) FPPD should have returned to the main playlist at the correct index and continued.
00:27 - FPPD Restarts - realises it should be playing the playlist due to the time and plays it, hence returning to the top of the playlist, playing 'Into the Unknown' a multitude more times than it should be.

**Additional Attachments**
[FPP Issue.mp4.zip](https://github.com/FalconChristmas/fpp/files/5662406/FPP.Issue.mp4.zip)
[FPP_Logs_20201208.2300.zip](https://github.com/FalconChristmas/fpp/files/5662441/FPP_Logs_20201208.2300.zip)

**Comment by cpinkham** [comment]
Can you disable MQTT and see if the problem goes away?  What if you set your MQTT publish frequency to 0 but leave MQTT on to indicate that it should only publish on demand instead of once per second.

It appears that the issue is happening inside Playlist::GetMqttStatusJSON().  If disabling MQTT fixes the issue then we may need to have you get another log dump with Debug level logging turned on for playlists and MQTT.

**Comment by cpinkham** [comment]
We have a good idea what this is and are trying to figure out the best solution.  If you can run the test above, that would help confirm.

**Comment by weazel91** [comment]
Hey!

I can confirm that last night I completely disabled MQTT and that sorted the issue. I've since re-imaged my home assistant instance so can't setup MQTT in the short term to test the publish frequency. If you require MQTT to be re-enabled to sort this bug so as you can get logs then let me know and I'll spin up another Home Assistant. 

Chris

**Comment by cpinkham** [comment]
I think we have a pretty good idea what is happening, we are just deciding what is the best solution.  Thanks for testing.  We'll update the ticket once we have a fix in git.

**Comment by mikem30303** [comment]
Hi, I have been fighting an issue that I believe I have finally locked down the problem.  MQTT.  I have wiped the Pi 3 times now with a fresh FPP install.  I now know that as soon as I enable MQTT, one of my controllers stops working.   Fortunately, I can remove the server IP from the MQTT settings, and everything seems to be working normally again.  I was hoping to turn off outside lighting automations when fpp kicks off but I think this needs to be put on hold.  Has there been any progress on this?  I'm using the latest fpp 6.2.  Thanks!
